### PR TITLE
feat(logs): add unit parameter support for MetricFilter

### PR DESCRIPTION
**Pull request by robertd**
----

Closes https://github.com/aws/aws-cdk/issues/23022

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-logs/lib/metric-filter.ts [hunk 1]
```diff
@@ -14,6 +14,146 @@ export interface MetricFilterProps extends MetricFilterOptions {
...
+
+  /**
+   * Megabytes
+   */
+  MEGABYTES = 'Megabytes',
+
+  /**
+   * Megabytes per second
+   */
+  MEGYTES_PER_SECOND = 'Megabytes/Second',
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-logs/lib/metric-filter.ts:109]:
```suggestion
  MEGABYTES_PER_SECOND = 'Megabytes/Second',
```

File: packages/@aws-cdk/aws-logs/lib/metric-filter.ts [hunk 2]
```diff
@@ -14,6 +14,146 @@ export interface MetricFilterProps extends MetricFilterOptions {
...
+
+  /**
+   * Terabits per second
+   */
+  TERABITS_PER_SECOND = 'Terabits/Second',
+
+  /**
+   * Terabytes
+   */
+  TERRABYTES = 'Terabytes',
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-logs/lib/metric-filter.ts:149]:
```suggestion
  TERABYTES = 'Terabytes',
```

**Review by comcalvi — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-logs/lib/metric-filter.ts (hunk 2 above)
Comment (robertd) [reply packages/@aws-cdk/aws-logs/lib/metric-filter.ts:149]: Whoops...

File: packages/@aws-cdk/aws-logs/lib/metric-filter.ts (hunk 1 above)
Comment (robertd) [reply packages/@aws-cdk/aws-logs/lib/metric-filter.ts:109]: Whoops...

**Comment by robertd** [comment]
@comcalvi PR is ready for review. Thanks.

**Comment by robertd** [comment]
@comcalvi Can you reopen this PR?

**Pull request closed without merging**